app/src/test/java/waiter/RouterTest.java

```diff
@@ -10,59 +10,59 @@ public class RouterTest {
     }
 
     @Test
     void getSimpleGetWithBodyReturnsOKRouteWithBody() {
         Request request = new Request("GET","/simple_get_with_body", "HTTP/1.1");
 
-        Route route = new Router().getRoute(request);
+        Response Response = new Router().getRequestedRoute(request);
 
-        assertEquals("HTTP/1.1", route.getProtocol());
-        assertEquals("200 OK", route.getStatus());
-        assertEquals("Content-Length: 11", route.getHeaders());
-        assertEquals("Hello world", route.getBody());
+        assertEquals("HTTP/1.1", Response.getProtocol());
+        assertEquals("200 OK", Response.getStatus());
+        assertEquals("Content-Length: 11", Response.getHeaders());
+        assertEquals("Hello world", Response.getBody());
     }
 
     @Test
     void headSimpleGetReturnsOKRouteWithoutBody() {
         Request request = new Request("HEAD","/simple_get", "HTTP/1.1");
 
-        Route route = new Router().getRoute(request);
+        Response Response = new Router().getRequestedRoute(request);
 
-        assertEquals("HTTP/1.1", route.getProtocol());
-        assertEquals("200 OK", route.getStatus());
-        assertEquals("Content-Length: 0", route.getHeaders());
-        assertEquals("", route.getBody());
+        assertEquals("HTTP/1.1", Response.getProtocol());
+        assertEquals("200 OK", Response.getStatus());
+        assertEquals("Content-Length: 0", Response.getHeaders());
+        assertEquals("", Response.getBody());
     }
 
     @Test
     void NoMethodOrUrlFoundReturnsInternalServerErrorRouteWithoutBody() {
         Request request = new Request("foo","/bar", "HTTP/1.1");
 
-        Route route = new Router().getRoute(request);
+        Response Response = new Router().getRequestedRoute(request);
 
-        assertEquals("HTTP/1.1", route.getProtocol());
-        assertEquals("404 Not Found", route.getStatus());
-        assertEquals("Content-Length: 0", route.getHeaders());
-        assertEquals("", route.getBody());
+        assertEquals("HTTP/1.1", Response.getProtocol());
+        assertEquals("404 Not Found", Response.getStatus());
+        assertEquals("Content-Length: 0", Response.getHeaders());
+        assertEquals("", Response.getBody());
     }
 
     @Test
     void NoUrlFoundReturnsNotFoundRouteWithoutBody() {
         Request request = new Request("HEAD","/bar", "HTTP/1.1");
 
-        Route route = new Router().getRoute(request);
+        Response Response = new Router().getRequestedRoute(request);
 
-        assertEquals("HTTP/1.1", route.getProtocol());
-        assertEquals("404 Not Found", route.getStatus());
-        assertEquals("Content-Length: 0", route.getHeaders());
-        assertEquals("", route.getBody());
+        assertEquals("HTTP/1.1", Response.getProtocol());
+        assertEquals("404 Not Found", Response.getStatus());
+        assertEquals("Content-Length: 0", Response.getHeaders());
+        assertEquals("", Response.getBody());
```

Comment:
What happens if you hit the resource in Postman or in the browser, would it just give you a blank page? I wonder if we should return a body that tells the user what happened.

Reply:
It would indeed just give a blank page, I could return a body that says something along the lines of `Resource could not be found, please try again`

Reply:
@matteeyao @elijahsullivan what do you all think?

Reply:
@mrfishball Agreed

Reply:
![404](https://user-images.githubusercontent.com/6895934/174850364-43d24762-8356-41b6-a3a1-867311da5812.png)

Here's the default 404 response from Next.js as an example.

Reply:
Not nearly as fancy (I can look into this if desired), but addressed in [this commit](https://github.com/jonmanahan/waiter/pull/18/commits/9c79171c8dc6dfaefd49fa6020893886b000ccad)